ni_measurementlink_service/_internal/parameter/metadata.py

```diff
@@ -50,51 +50,95 @@ def validate_default_value_type(parameter_metadata: ParameterMetadata) -> None:
-                        f"Unexpected element of type {type(element)} in the default value for '{display_name}'. Expected element type: {expected_element_type}."
-                    )
+        _validate_default_value_type_for_repeated_type(
+            default_value,
+            expected_type,
+            expected_element_type,
+            enum_values_annotation,
+            display_name,
+        )
     else:
-        if has_enum_values_annotation:
-            if not _is_default_enum_match_annotations(default_value, enum_values_annotation):
-                raise TypeError(
-                    f"The default value, `{default_value}`, for '{display_name}' is not valid. Expected values: `{enum_values_annotation}`."
-                )
-        else:
-            if not isinstance(default_value, expected_type):
-                raise TypeError(
-                    f"Unexpected type {type(default_value)} in the default value for '{display_name}'. Expected type: {expected_type}."
-                )
+        _validate_default_value_type_for_scalar_type(
+            default_value, expected_type, enum_values_annotation, display_name
+        )
     return None
 
 
-def try_get_enum_values_annotation(parameter_metadata: ParameterMetadata) -> Tuple[bool, str]:
+def _validate_default_value_type_for_scalar_type(
+    default_value: Any, expected_type: type, enum_values_annotation: str, display_name: str
+) -> None:
+    """Validate and raise exception if the default value does not match the type info."""
+    if enum_values_annotation:
+        user_enum_dict = json.loads(enum_values_annotation.replace("'", '"'))
+        _validate_default_value_type_for_enum_type(
+            default_value, user_enum_dict, enum_values_annotation, display_name
+        )
+    else:
+        _validate_default_value_type_for_basic_type(default_value, expected_type, display_name)
+
+
+def _validate_default_value_type_for_repeated_type(
+    default_value: Iterable[Any],
+    expected_type: type,
+    expected_element_type: type,
+    enum_values_annotation: str,
+    display_name: str,
+) -> None:
+    """Validate and raise exception if the default value does not match the type info."""
+    if not isinstance(default_value, expected_type):
+        raise TypeError(
+            f"Unexpected type {type(default_value)} in the default value for '{display_name}'. Expected type: {expected_type}."
+        )
+
+    if enum_values_annotation:
+        user_enum_dict = json.loads(enum_values_annotation)
+        for element in default_value:
+            _validate_default_value_type_for_enum_type(
+                element, user_enum_dict, enum_values_annotation, display_name
+            )
+    else:
+        for element in default_value:
+            _validate_default_value_type_for_basic_type(
+                element, expected_element_type, display_name
+            )
+
+
+def _validate_default_value_type_for_basic_type(
+    default_value: Any,
```

Comment:
FYI: https://mypy.readthedocs.io/en/stable/dynamic_typing.html#any-vs-object

Reply:
Switched to using `object` for these private validation helpers instead of `Any`.